outline/shadowsocks/connectivity.go

```diff
@@ -32,12 +32,7 @@ const reachabilityTimeout = 10 * time.Second
 // the current network. Parallelizes the execution of TCP and UDP checks, selects the appropriate
 // error code to return accounting for transient network failures.
 // Returns an error if an unexpected error ocurrs.
-func CheckConnectivity(host string, port int, password, cipher string) (int, error) {
-	client, err := shadowsocks.NewClient(host, port, password, cipher)
-	if err != nil {
-		// TODO: Inspect error for invalid cipher error or proxy host resolution failure.
-		return Unexpected, err
-	}
+func CheckConnectivity(client *Client) (int, error) {
 	tcpChan := make(chan error)
 	// Check whether the proxy is reachable and that the client is able to authenticate to the proxy
 	go func() {
```

Comment:
Not for this PR, but we should fail the connectivity test if TCP fails. Currently it will succeed if UDP succeeds.